src/lib.rs

```diff
@@ -46,8 +50,11 @@ pub fn edit_distance(a: &str, b: &str) -> i32 {
                 // match or substitution
                 matrix[i][j] + if ca == cb { 0 } else { 1 }];
             matrix[i+1][j+1] = *alternatives.iter().min().unwrap();
+            j = j + 1;
         }
+        j = 0;
+        i = i + 1;
```

Comment:
Would [enumerate](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.enumerate) help? It's basically like rubys `each_with_index`.